### ISSUE TITLE
Use of signac.config in signac-flow

### ISSUE THREAD

**Issue by b-butler** [glotzerlab/signac-flow#643]
In https://github.com/glotzerlab/signac, we are looking to remove or significantly reduce our config Python API. To do so, we need to understand where and how we use this API in flow. Currently we use the config API to,
1. load in/require a specific key `config.load_config()["flow"]`
2. find environment specific settings (e.g. account name) `config.read_config_file(config.FN_CONFIG)`
3. Allow templates to require configuration values through the function using 1. We use this to require account names.

All the use cases we have could be achieved by a function which gets the configuration immutably. We could also make the configuration restricted to a project instance, but we would need to change the code to make getting environment specific configuration a project task.

**Comment by b-butler** [comment]
I missed one use, and that is to optionally import our packaged environments. I am not sure if this is something we want to support or not, but we can't wait to get a Project instance in this case.

**Comment by vyasr** [comment]
I haven't looked at this recently, but IIRC yes we do need the ability to get config information outside of a specific project's context for exactly the use-case that your last message raises. However, it sounds like what this basically boils down to is that all we need is for signac to promise to use a certain file format for the configuration, right? As long as the file format is fixed, signac-flow should be able to rely on reading it using a standard library for that format, and it doesn't need any special handling from within signac. Does that sound right to you?

**Comment by vyasr** [comment]
Yeah that's a good point. So concretely, it sounds like we have two options:
1. signac promises:
    1. A specific file format for the config. That specification should be part of the schema and can change when signac undergoes a schema version update.
    2. Provide a public API to query the list of config files (in order of priority).
2. signac provides:
    1. A function that returns the config data in a dict. This function would have to work outside of a signac project directory so that you could access the global config information without the local.

Does that sound right to you?

**Comment by vyasr** [comment]
CC @bdice

**Comment by vyasr** [comment]
> Is this function public or private (internally stable for use by other signac projects)?

Public. IMO we should discourage usage of any non-public attributes of signac in signac-* packages.

**Comment by vyasr** [comment]
This issue can be closed after #649 is merged. It is resolved by a combination of #649, #652, and #653.

All usage of the config is now restricted to project-specific data, so we no longer require the ability to access the config except through the project. As a result, once #649 is merged signac-flow will rely only on `FlowProject.config` (inherited from `signac.Project.config`.
